app/services/data_migrations/fix_lat_long_flip_flops.rb

```diff
@@ -4,91 +4,21 @@ class FixLatLongFlipFlops
     MANUAL_RUN = true
 
     def change
-      audits = Audited::Audit.where(
-        "auditable_type = 'Provider' AND " \
-        "action = 'update' AND " \
-        "audited_changes ?& array['latitude', 'longitude'] AND " \
-        "(audited_changes - 'latitude') - 'longitude' = '{}'",
-      ).order(created_at: :asc)
-
-      deleted_audit_count = 0
-      total_audits = audits.count
-
       log("Before: audits table size is #{audits_table_size}")
 
-      providers.find_each do |p|
-        log("Cleaning up provider #{p.id} (#{p.name_and_code})")
-
-        audits_for_provider = audits.where(auditable_id: p.id)
-        audits_for_provider_count_before_cleanup = audits.where(auditable_id: p.id).count
-
-        if audits_for_provider.empty?
-          log("No audits to clean up for provider #{p.id}")
-          next
-        end
-
-        # if we already had lat/lng when we made the first spurious update, delete every audit
-        if audits_for_provider.first.audited_changes['latitude'].last.nil?
-          delete_audits(
-            'lat/long-only audits for this provider (all of them) becase lat/long was set beforehand',
-            audits_for_provider,
-          )
-
-        # otherwise, delete all the audits except the one that set it for the first time
-        else
-
-          delete_audits(
-            'audits which set the lat/long to nil',
-            audits_for_provider.where("audited_changes#>>'{longitude, 1}' is null"),
-          )
-
-          delete_audits(
-            'audits which repeatedly set the lat/long to the same value',
-            audits_for_provider.where("audited_changes#>>'{longitude, 0}' is null").offset(1),
-          )
-        end
-
-        deleted_audit_count += (audits_for_provider_count_before_cleanup - audits_for_provider.count)
-      end
-
-      log("Deleted #{deleted_audit_count} lat/long audits out of #{total_audits}") unless dry_run?
+      Audited::Audit.connection.execute("DELETE FROM audits WHERE (auditable_type = 'Provider' AND action = 'update' AND audited_changes ?& array['latitude', 'longitude'] AND (audited_changes - 'latitude') - 'longitude' = '{}')")
```

Comment:
That's right, it will only delete audits for Provider updates which only change latitude and longitude. Which means any lat/lon changes bundled with other changes may be preserved. But running this query as is will remove about 1.4 million audit records and reducing the number of rows is really what we're after.